hypothesis-python/src/hypothesis/internal/conjecture/datatree.py

```diff
@@ -333,25 +422,37 @@ def draw_bits(self, n_bits, forced, value):
                 node.split_at(i)
                 assert i == len(node.values)
                 new_node = TreeNode()
-                branch = node.transition
-                branch.children[value] = new_node
+                node.transition.children[value] = new_node
                 self.__current_node = new_node
                 self.__index_in_current_node = 0
         else:
             trans = node.transition
             if trans is None:
-                node.bit_lengths.append(n_bits)
+                node.ir_types.append(ir_type)
+                node.kwargs.append(kwargs)
                 node.values.append(value)
                 if forced:
                     node.mark_forced(i)
+                # generate_novel_prefix assumes the following invariant: any one
+                # of the series of draws in a particular node can vary. This is
+                # true if all nodes have more than one possibility, which was
+                # true when the underlying representation was bits (lowest was
+                # n=1 bits with m=2 choices).
+                # However, with the ir, e.g. integers(0, 0) has only a single
+                # value. To retain the invariant, we forcefully split such cases
+                # into a transition.
```

Comment:
Alternatively, we could refactor things to omit exact-value 'psudeo-choices' from the IR - we already do that for `st.just()` iirc, and it would be perfectly precise if we did so in `ConjectureData`.

Reply:
I like that proposal better, and I'll throw in an `assert max_children > 1` in case this comes up for other IR nodes (strings with `max_size = 0`, probably?). I was a little concerned about the performance implications of calculating the number of children so frequently here anyway...

Reply:
I've done this in 4ab240dd1773a3ce80935db842dee53970d784cc.

I think this is the right thing to do. We get performance benefits here because a pseudo-choice can never vary and is pointless to try shrinking / track in the datatree or in examples. Maybe we lose out on detecting some flakiness by not tracking it?

But it is nontrivial to get right — see TODOs in above commit. If this was something every strategy had to keep in mind I would say it is not worth it. But since it's just our IR nodes, it probably is worthwhile.